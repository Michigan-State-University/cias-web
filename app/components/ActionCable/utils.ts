import { TokenRefresher } from '@anycable/core/create-cable';

import { URIEncodeObject } from 'utils/uriUtils';
import LocalStorageService from 'utils/localStorageService';

export const createCableUrl = (
  headers: Nullable<Record<string, string>>,
): Nullable<string> => {
  if (!headers) return null;
  const { 'Access-Token': AccessToken, Uid, Client } = URIEncodeObject(headers);
  if (!AccessToken || !Uid || !Client) return null;
  return `${process.env.WEBSOCKET_URL}?access_token=${AccessToken}&uid=${Uid}&client=${Client}`;
};

export const tokenRefresher: TokenRefresher = async (transport) => {
  const url = createCableUrl(LocalStorageService.getHeaders());
  if (url) {
    transport.setURL(url);
  }
};

export const getUidFromCableUrl = (cableUrl: string) =>
  new URL(cableUrl).searchParams.get('uid');
