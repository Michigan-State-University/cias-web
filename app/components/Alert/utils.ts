import { colors, themeColors } from 'theme';

import { AlertType } from './types';

export const getBackgroundColorByType = (type: AlertType) => {
  switch (type) {
    case AlertType.WARNING: {
      return colors.chardonnay;
    }
    case AlertType.INFO:
    default: {
      return themeColors.highlight;
    }
  }
};
