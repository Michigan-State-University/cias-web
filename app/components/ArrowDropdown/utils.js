import { createContext } from 'react';

export const ArrowDropdownContext = createContext({ ref: null });
