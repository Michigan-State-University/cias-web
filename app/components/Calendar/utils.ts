import {
  MAX_VISIBLE_EVENTS_SMALL,
  MAX_VISIBLE_EVENTS_MEDIUM,
  MAX_VISIBLE_EVENTS_LARGE,
} from './constants';

export const getNumberOfEventsVisible = (calendarRows: number) => {
  switch (true) {
    case calendarRows === 5:
      return MAX_VISIBLE_EVENTS_MEDIUM;
    case calendarRows > 5:
      return MAX_VISIBLE_EVENTS_SMALL;
    case calendarRows < 5:
      return MAX_VISIBLE_EVENTS_LARGE;
    default:
      return MAX_VISIBLE_EVENTS_MEDIUM;
  }
};
