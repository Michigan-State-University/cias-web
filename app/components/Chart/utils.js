import regression from 'regression';
import range from 'lodash/range';

import { TREND_LINE_DATA_KEY } from './constants';

export const generateBarChartTrendData = (data, yKey) => {
  const trendParams = range(data.length).map(index => [
    index,
    data[index][yKey],
  ]);
  const rawTrendData = regression.polynomial(trendParams, { order: 3 });

  return rawTrendData.points.map(([, y]) => ({ [TREND_LINE_DATA_KEY]: y }));
};
