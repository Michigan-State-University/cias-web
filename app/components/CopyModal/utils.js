import { createContext } from 'react';

export const CopyModalContext = createContext({ interventionStatusFilter: [] });

export const InfiniteScrollContext = createContext({
  selectedItem: null,
  changeViewAction: null,
  selectAction: null,
  disableCopy: null,
  listIcon: null,
  disabledItemsIds: null,
});
