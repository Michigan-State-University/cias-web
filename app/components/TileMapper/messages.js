import { defineMessages } from 'react-intl';

export const scope = 'app.containers.ParticipantDashboard';

export default defineMessages({
  noResults: {
    id: `${scope}.noResults`,
    defaultMessage: 'No Results',
  },
});
