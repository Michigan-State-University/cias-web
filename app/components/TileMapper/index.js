import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import Column from 'components/Column';
import H2 from 'components/H2';
import Loader from 'components/Loader';

import isNullOrUndefined from 'utils/isNullOrUndefined';
import messages from './messages';

const TileMapper = ({ items, component, loading }) => (
  <>
    {loading && <Loader />}
    {items && items.map(component)}
    {!loading && (isNullOrUndefined(items) || items.length === 0) && (
      <Column width="100%" align="center" mt={50}>
        <H2>
          <FormattedMessage {...messages.noResults} />
        </H2>
      </Column>
    )}
  </>
);

TileMapper.propTypes = {
  items: PropTypes.array,
  component: PropTypes.func,
  loading: PropTypes.bool,
};

export default TileMapper;
