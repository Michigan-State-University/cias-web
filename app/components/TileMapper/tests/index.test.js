import React from 'react';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'i18n';

import TileMapper from 'components/TileMapper/index';
import { createReport } from 'utils/reducerCreators';

describe('<TileMapper />', () => {
  const mockReports = [createReport()];

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <TileMapper
            items={mockReports}
            component={({ id, title }) => <h3 key={id}>{title}</h3>}
          />
        </MemoryRouter>
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <TileMapper
            items={mockReports}
            component={({ id, title }) => <h3 key={id}>{title}</h3>}
          />
        </MemoryRouter>
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render and match the snapshot - empty', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemoryRouter>
          <TileMapper
            items={[]}
            component={({ id, title }) => <h3 key={id}>{title}</h3>}
          />
        </MemoryRouter>
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
