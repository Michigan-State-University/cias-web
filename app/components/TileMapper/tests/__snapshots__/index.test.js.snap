// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileMapper /> Should render and match the snapshot - empty 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin-top: 50px;
}

.c1 {
  margin: 0;
  font-family: DM Sans;
  font-size: 18px;
  line-height: 130%;
  color: #2F3850;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:960px) {

}

@media only screen and (min-width:1170px) {

}

@media only screen and (min-width:1360px) {

}

<div>
  <div
    class="c0"
    width="100%"
  >
    <h2
      class="c1"
    >
      <span>
        No reports available at the moment
      </span>
    </h2>
  </div>
</div>
`;

exports[`<TileMapper /> Should render and match the snapshot 1`] = `
<div>
  <h3>
    Report: 0. part - Results
  </h3>
</div>
`;
